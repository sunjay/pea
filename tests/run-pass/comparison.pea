fn compare(a: i64, b: i64) {
    println!(b"a == b");
    println!(a == b);
    println!(b"a != b");
    println!(a != b);
    println!(b"a > b");
    println!(a > b);
    println!(b"a >= b");
    println!(a >= b);
    println!(b"a < b");
    println!(a < b);
    println!(b"a <= b");
    println!(a <= b);
}

fn main() {
    println!(b"WOW LOOK AT THESE COMPARISONS");

    println!(b"\n1 vs. 1\n");
    compare(1, 1);

    println!(b"\n1 vs. 2\n");
    compare(1, 2);

    println!(b"\n300 vs. -300\n");
    compare(300, -300);

    println!(b"\n\"abc\" vs. \"abc\"\n");
    let a = b"abc";
    let b = a;
    println!(b"a == b");
    println!(a == b);
    println!(b"a != b");
    println!(a != b);
}
