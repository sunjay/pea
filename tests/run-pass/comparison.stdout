WOW LOOK AT THESE COMPARISONS

1 vs. 1

a == b
true
a != b
false
a > b
false
a >= b
true
a < b
false
a <= b
true

1 vs. 2

a == b
false
a != b
true
a > b
false
a >= b
false
a < b
true
a <= b
true

300 vs. -300

a == b
false
a != b
true
a > b
true
a >= b
true
a < b
false
a <= b
false

"abc" vs. "abc"

a == b
true
a != b
false
